sdk/common/src/main/java/io/opentelemetry/sdk/resources/ResourceBuilder.java

```diff
@@ -0,0 +1,177 @@
+   *
+   * @return this Builder
+   */
+  public ResourceBuilder put(String key, boolean value) {
+    if (key != null) {
+      attributesBuilder.put(key, value);
+    }
+    return this;
+  }
+
+  /**
+   * Puts a String array attribute into this.
+   *
+   * <p>Note: It is strongly recommended to use {@link #put(AttributeKey, Object)}, and pre-allocate
+   * your keys, if possible.
+   *
+   * @return this Builder
+   */
+  public ResourceBuilder put(String key, String... values) {
+    if (key != null && values != null) {
+      attributesBuilder.put(key, values);
+    }
+    return this;
+  }
+
+  /**
+   * Puts a Long array attribute into this.
+   *
+   * <p>Note: It is strongly recommended to use {@link #put(AttributeKey, Object)}, and pre-allocate
+   * your keys, if possible.
+   *
+   * @return this Builder
+   */
+  public ResourceBuilder put(String key, long... values) {
+    if (key != null && values != null) {
+      attributesBuilder.put(key, values);
+    }
+    return this;
+  }
+
+  /**
+   * Puts a Double array attribute into this.
+   *
+   * <p>Note: It is strongly recommended to use {@link #put(AttributeKey, Object)}, and pre-allocate
+   * your keys, if possible.
+   *
+   * @return this Builder
+   */
+  public ResourceBuilder put(String key, double... values) {
+    if (key != null && values != null) {
+      attributesBuilder.put(key, values);
+    }
+    return this;
+  }
+
+  /**
+   * Puts a Boolean array attribute into this.
+   *
+   * <p>Note: It is strongly recommended to use {@link #put(AttributeKey, Object)}, and pre-allocate
+   * your keys, if possible.
+   *
+   * @return this Builder
+   */
+  public ResourceBuilder put(String key, boolean... values) {
+    if (key != null && values != null) {
+      attributesBuilder.put(key, values);
+    }
+    return this;
+  }
+
+  /** Puts a {@link AttributeKey} with associated value into this. */
+  public <T> ResourceBuilder put(AttributeKey<T> key, T value) {
+    if (key != null && key.getKey() != null && key.getKey().length() > 0 && value != null) {
+      attributesBuilder.put(key, value);
+    }
+    return this;
+  }
+
+  /** Puts a {@link AttributeKey} with associated value into this. */
+  public ResourceBuilder put(AttributeKey<Long> key, int value) {
+    if (key != null && key.getKey() != null) {
+      attributesBuilder.put(key, value);
+    }
+    return this;
+  }
+
+  /** Puts all {@link Attributes} into this. */
+  public ResourceBuilder putAll(Attributes attributes) {
+    if (attributes != null) {
+      attributesBuilder.putAll(attributes);
+    }
+    return this;
+  }
+
+  /** Puts all attributes from {@link Resource} into this. */
+  public ResourceBuilder putAll(Resource resource) {
+    if (resource != null) {
+      attributesBuilder.putAll(resource.getAttributes());
+    }
+    return this;
+  }
+
+  /** Create the {@link Resource} from this. */
+  public Resource build() {
+    return new AutoValue_Resource(attributesBuilder.build());
```

Comment:
should this just call `Resource.create()` rather than having an extra reference to the autovalue implementation ?